### DISCUSSION TITLE
When using create, why is values always an Optional<> of CreationAttributes?

### DISCUSSION THREAD

**Discussion by karpour** [sequelize/sequelize#16015] [Help & Questions] [answered]
I have the following Typescript code:

```typescript
export class PlaylistModel extends Model<InferAttributes<PlaylistModel>, InferCreationAttributes<PlaylistModel>> {
  declare id: CreationOptional<number>;
  declare name: string;
  declare type: PlaylistType;
  declare algorithm: PlaylistAlgorithmType;
  declare createdAt: CreationOptional<Date>;
  declare updatedAt: CreationOptional<Date>;
  [...]
}
[...]
```

I would expect the create method to require all non-optional attributes, making the following input invalid (it results in a run-time error of course, but on a type level it's valid.

```typescript
PlaylistModel.create({}); // No type error but runtime error
```

The expected minimal valid input should be with the `name`, `type` and `algorithm` fields set, as follows.

```typescript
// Expected minimal valid input
PlaylistModel.create({
  name: "Name",
  type: 'live',
  algorithm: "weightedshuffle",
});
```

I can see that in create(), values is set to `Optional<InferCreationAttributes<PlaylistModel>>` while for type safety I would like it to be just `InferCreationAttributes<PlaylistModel>`

Am I using something the wrong way?

**Comment by ephys** [comment]
Which version of TypeScript and Sequelize are you using? That Optional you're seeing is not complete, it only marks properties that accept null or are tagged as `CreationOptional` as optional

**Reply by karpour** [reply, marked as answer]
Hi, thank you for your reply. I tried recreating this issue from a while ago, but was unable to recreate it using the same code posted above.
Now `values` is of type `Optional<InferCreationAttributes<PlaylistModel, { omit: never; }>, NullishPropertiesOf<InferCreationAttributes<PlaylistModel, { omit: never; }>>>` as expected.

It might have been an issue with the project setup, Typescript version or some import mishap. Please consider this issue closed.
